From 7185248a85fcd2865fe16229a4b8401fb0432fb2 Mon Sep 17 00:00:00 2001
From: Mark Asselstine <mark.asselstine@windriver.com>
Date: Sat, 9 Sep 2023 19:04:38 +0000
Subject: [PATCH] stdlib: Makefile: always assume long shebang

There is a TODO to handle long shebangs, but for now it just defaults
to assume they will not be long. Since we will almost certainly hit
this with our native package let's default instead to assume long
shebang.

Signed-off-by: Mark Asselstine <mark.asselstine@windriver.com>
---
 stdlib/Makefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/stdlib/Makefile b/stdlib/Makefile
index ac0ef4c..07c5240 100644
--- a/stdlib/Makefile
+++ b/stdlib/Makefile
@@ -142,9 +142,10 @@ else
 endif
 
 # TODO This does not take long shebangs into account (since TARGET_BINDIR is not
-#      yet processed by configure)
+#      yet processed by configure) so always assume long shebang
 target_%: $(ROOTDIR)/Makefile.config Makefile
-	echo '#!$(TARGET_BINDIR)/ocamlrun$(subst camlheader,,$*)' > $@
+	echo '#!/bin/sh' > $@
+	echo 'exec "$(TARGET_BINDIR)/ocamlrun$(subst camlheader,,$*)" "$$0" "$$0"' >> $@
 
 camlheader_ur: Makefile
 	echo '#!' | tr -d '\012' > $@
-- 
2.35.5

